contracts/ChargedParticles.sol

```diff
@@ -643,6 +644,9 @@ contract ChargedParticles is Initializable, IParticleManager, BaseRelayRecipient
         bytes16 _assetPairId = typeAssetPairId[_typeId];
         _collectAssetToken(_msgSender(), _assetPairId, _assetAmount);
 
+        // Approve escrowMgr to spend _assetAmount
+        IApproval(escrowMgr.getAssetTokenAddress(_assetPairId)).approve(address(escrowMgr), _assetAmount);
```

Comment:
We should do this in `registerEscrowManager` with a value of `uint(-1)` for infinite approval